### PR TITLE
fix(pkg/nar): loosen node name restrictions

### PR DESCRIPTION
**Pull request by flokli**
Nix paths have much stronger requirements than regular paths. The only
thing we don't want to see is a / in the node name.

cc: @puckipedia

### CODE REVIEW

File: pkg/nar/header.go [hunk 1]
```diff
@@ -15,6 +17,44 @@ type Header struct {
...
+
+		if strings.ContainsAny(h.Path, "\u0000") {
+			return fmt.Errorf("path contains null bytes")
+		}
+	}
+
+	// Regular files and directories may not have LinkTarget set.
+	if h.Type == TypeRegular || h.Type == TypeDirectory {
+		if h.LinkTarget != "" {
+			if h.LinkTarget != "" {
```
Comment (andir) [inline pkg/nar/header.go:38]: I think you can drop one of these checks as they are equivalent

File: pkg/nar/header.go [hunk 2]
```diff
@@ -15,6 +17,44 @@ type Header struct {
...
+	if h.Type == TypeDirectory || h.Type == TypeSymlink {
+		if h.Size != 0 {
+			return fmt.Errorf("type is %v, but Size is not 0", h.Type.String())
+		}
+
+		if h.Executable {
+			return fmt.Errorf("type is %v, but Executable is true", h.Type.String())
+		}
+	}
+
```
Comment (andir) [inline pkg/nar/header.go:54]: Currently empty link targets are accepts for `TypeSymlink`

File: pkg/nar/header.go (hunk 1 above)
Comment (flokli) [reply pkg/nar/header.go:38]: Oops, this was some reorganization error. I dropped it here and in ls/list.go.

File: pkg/nar/header.go (hunk 2 above)
Comment (flokli) [reply pkg/nar/header.go:52]: nice catch! fixed.

**Review by andir — APPROVED** [review]

**Pull request merged**